codalab/apps/teams/views.py

```diff
@@ -0,0 +1,324 @@
+from django.utils.timezone import now
+from django.contrib.auth import get_user_model
+from django.shortcuts import render, get_object_or_404
+from django.views.generic import DetailView, TemplateView
+from django.core.urlresolvers import reverse
+
+from apps.web.models import Competition, ParticipantStatus, CompetitionSubmission, get_current_phase
+from apps.web.views import LoginRequiredMixin
+
+from .models import Team, TeamStatus, TeamMembership, TeamMembershipStatus, get_user_requests, get_competition_teams, get_user_team, get_allowed_teams, get_team_pending_membership, get_competition_user_pending_teams, get_team_submissions_inf
+from apps.teams import forms
+from django.views.generic import View, TemplateView, DetailView, ListView, FormView, UpdateView, CreateView, DeleteView
+from extra_views import CreateWithInlinesView, UpdateWithInlinesView, InlineFormSet, NamedFormsetsMixin
+
+from django.db.models import Q
+from django.http import Http404, QueryDict
+from django.http import HttpResponse, HttpResponseRedirect, HttpResponseForbidden
+from django.http import StreamingHttpResponse
+
+User = get_user_model()
+
+class TeamDetailView(LoginRequiredMixin, TemplateView):
+    # Serves the table of submissions in the Participate tab of a competition.
+    # Requires an authenticated user who is an approved participant of the competition.
+    template_name = 'teams/team_info.html'
+    form_class = forms.TeamMembershipForm
+
+    def post(self, request, *args, **kwargs):
```

Comment:
Not sure, I'm not expert on DJango. I followed examples I found. It works, but probably is not the best way to do that.